### DISCUSSION TITLE
Error starting Dataline binary - 'NoneType' object has no attribute 'id'

### DISCUSSION THREAD

**Discussion by tlawrence-TAG** [RamiAwar/dataline#343] [Q&A] [answered]
pydantic/_internal/_fields.py:200: UserWarning: Field name "api_key" in "ShouldGenerateChartCall" shadows an attribute in parent "OpenAIExtractor[ShouldGenerateChart]"
pydantic/_internal/_fields.py:200: UserWarning: Field name "base_url" in "ShouldGenerateChartCall" shadows an attribute in parent "OpenAIExtractor[ShouldGenerateChart]"
pydantic/_internal/_fields.py:200: UserWarning: Field name "api_key" in "GenerateChartCall" shadows an attribute in parent "OpenAIExtractor[GeneratedChart]"
pydantic/_internal/_fields.py:200: UserWarning: Field name "base_url" in "GenerateChartCall" shadows an attribute in parent "OpenAIExtractor[GeneratedChart]"
pydantic/_internal/_fields.py:200: UserWarning: Field name "api_key" in "ConversationTitleGenerator" shadows an attribute in parent "OpenAICall"
pydantic/_internal/_fields.py:200: UserWarning: Field name "base_url" in "ConversationTitleGenerator" shadows an attribute in parent "OpenAICall"
INFO:     Started server process [13927]
INFO:     Waiting for application startup.
INFO:alembic.runtime.migration:Context impl SQLiteImpl.
INFO:alembic.runtime.migration:Will assume non-transactional DDL.
ERROR:    Traceback (most recent call last):
  File "starlette/routing.py", line 677, in lifespan
  File "contextlib.py", line 204, in __aenter__
  File "dataline/main.py", line 61, in lifespan
**AttributeError: 'NoneType' object has no attribute 'id'**

ERROR:    Application startup failed. Exiting.
/usr/bin/xdg-open: 882: x-www-browser: not found
/usr/bin/xdg-open: 882: firefox: not found
/usr/bin/xdg-open: 882: iceweasel: not found
/usr/bin/xdg-open: 882: seamonkey: not found
/usr/bin/xdg-open: 882: mozilla: not found
/usr/bin/xdg-open: 882: epiphany: not found
/usr/bin/xdg-open: 882: konqueror: not found
/usr/bin/xdg-open: 882: chromium: not found
/usr/bin/xdg-open: 882: chromium-browser: not found
/usr/bin/xdg-open: 882: google-chrome: not found
/usr/bin/xdg-open: 882: www-browser: not found
/usr/bin/xdg-open: 882: links2: not found
/usr/bin/xdg-open: 882: elinks: not found
/usr/bin/xdg-open: 882: links: not found
/usr/bin/xdg-open: 882: lynx: not found
/usr/bin/xdg-open: 882: w3m: not found
xdg-open: no method available for opening 'http://localhost:7377'

I downloaded the linux distro and am running this on WSL2. Any help would be appreciated!

**Comment by RamiAwar** [comment, marked as answer]
Hey @tlawrence-TAG ! It looks like DataLine is trying to open http://localhost:7377 in a web browser but is unable to find any web browsers.

We use `webbrowser` from the Python standard library.

I'm not sure if you have any browsers installed already, maybe you don't. If you do, then the issue is with the Python standard lib package not supporting that browser.

To solve this, I'd recommend you install a mainstream browser 😄 Not much we can do otherwise. DataLine should run regardless, it's just a webserver. If you realllyyy want to use an unsupported browser, you could use DataLine as a docker container then and open the URL yourself!

Lmk if this doesn't solve the problem and we can re-open this discussion 🙂

**Reply by tlawrence-TAG** [reply]
I was using the docker image. This happens when you start dataline - you can see that from my trace file in the discussion.

**Reply by RamiAwar** [reply]
I guess running from inside WSL is the culprit then. It doesn't seem able to hook to your main browser.

Hmmm... I guess we could allow failures on opening web browser since that's more a UX thing. So try to auto-open, if it fails, do nothing.

I'll include this in the next build. In the meantime, you could just run the binary on some platform or maybe the docker container from windows. Those should work fine!

**Reply by RamiAwar** [reply]
Opened #345 for this
